README.md

```diff
@@ -56,23 +58,29 @@ The plugin behavior can be controlled with the following list of settings
 - `platform_map` (dictionary), mapping of an **auto-detected** Netmiko platform to the **NetBox slug** name of your Platform. The dictionary should be in the format:
     ```python
     {
-      <Netmiko Platform>: <NetBox Slug> 
+      <Netmiko Platform>: <NetBox Slug>
+    }
+    ```
+- `onboarding_extensions_map` (dictionary), mapping of a NAPALM driver name to the loadable Python module used as an onboarding extension. The dictionary should be in the format:
+    ```python
+    {
+      <Napalm Driver Name>: <Loadable Python Module>
     }
     ```
 
 ## Usage
 
 ### Preparation
 
-To work properly the plugin needs to know the Site, Platform, Device Type, Device Role of each
-device as well as its primary IP address or DNS Name. It's recommended to create these objects in
-NetBox ahead of time and to provide them when you want to start the onboarding process.
+To properly onboard a device, the plugin needs to only know the Site as well as device's primary IP address or DNS Name.
 
 > For DNS Name Resolution to work, the instance of NetBox must be able to resolve the name of the
 > device to IP address.
 
+Providing other attributes (`Platform`, `Device Type`, `Device Role`) is optional - if any of these attributes is provided, plugin will use provided value for the onboarded device.
 If `Platform`, `Device Type` and/or `Device Role` are not provided, the plugin will try to identify these information automatically and, based on the settings, it can create them in NetBox as needed.
-> If the Platform is provided, it must contains a valid Napalm driver available to the worker in Python
+> If the Platform is provided, it must point the the existing NetBox Platform. NAPALM driver of this platform will be used only if it is defined for the platform in NetBox.
```

Comment:
"...it must point to an existing NetBox Platform." perhaps?

Reply:
the the the You are correct :-)